core/unix/os.c

```diff
@@ -3719,9 +3719,11 @@ dr_create_client_thread(void (*func)(void *param), void *arg)
 #        endif
     LOG(THREAD, LOG_ALL, 1, "dr_create_client_thread xsp=" PFX " dstack=" PFX "\n", xsp,
         get_clone_record_dstack(crec));
+    os_clone_pre(dcontext);
```

Comment:
But we're already doing os_switch_lib_tls to app a few lines above, so there is now redundancy we should alleviate.

Reply:
Wait, but this switches the "lib tls" (os_switch_*lib*_tls) right, i.e. "fs"? Not DR's "gs" (Linux X64).

Reply:
Like I said, this code is already doing os_switch_lib_tls, and you're adding code doing the exact same thing over again (one of the 2 things that os_clone_pre does).

Reply:
I see now! => redundant one removed.